Fix #3822: Remove maxLines contsraint's to enable long text to linebreak

KevinGitonga opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
  Fix #3822: This PR fixes Portuguese text cut off by completely removing maxLines contraint's used in the several variations of the onboarding_slide layouts on the title text . This allows for the text to easily break to the next line incase the text is very long to fit in a single line.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [X] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [X] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [X] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [X] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [X] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [X] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

## Phone screenshots
| Before Portrait            |  After Portrait |
:-------------------------:|:-------------------------:
<img src="https://user-images.githubusercontent.com/20886444/187904033-3c28fc3a-11d7-4a81-aca3-9b38b16bca6a.jpg" width="300"/> | <img src="https://user-images.githubusercontent.com/20886444/188426980-b1f5ab52-b785-421b-9c77-559c612e643e.jpeg" width="300"/>

| Before Landscape            |  After Landscape |
:-------------------------:|:-------------------------:
<img src="https://user-images.githubusercontent.com/20886444/187904625-62a49b0f-2be8-49f2-9907-d30e246bd8ac.jpg" width="300"/> | <img src="https://user-images.githubusercontent.com/20886444/187904632-17c799db-b81f-4db0-a402-a127a6b8834e.jpeg" width="300"/> 

## Tab screenshots
| Before Portrait            |  After Portrait |
:-------------------------:|:-------------------------:
<img src="https://user-images.githubusercontent.com/20886444/188436852-2b931fa0-fe50-41bc-9860-3d45660bc37e.png" width="300"/> | <img src="https://user-images.githubusercontent.com/20886444/188436043-1875b313-81b6-4fdd-92ea-d4dc9b5dbab8.png" width="300"/>

| Before Landscape            |  After Landscape |
:-------------------------:|:-------------------------:
<img src="https://user-images.githubusercontent.com/20886444/188436940-f83860dd-0b32-4f7b-8fc5-ede29538f39c.png" width="300"/> | <img src="https://user-images.githubusercontent.com/20886444/188436083-658f7be7-be87-4bf2-9b4a-5f59fd6624c9.png" width="300"/>

KevinGitonga committed:
Reduce MarginStart and MarginEnd.

BenHenning commented:
@KevinGitonga my only concern with the solution is it may not work for longer translations that we might encounter in the future. I wonder: is it reasonable to introduce line-wrapping here, instead, for cases when the title is too long?

@rt4914 curious about your thoughts here, too.

KevinGitonga committed:
Update maxLines to automatically to show extra text on new line.

KevinGitonga commented:
@BenHenning it actually makes sense to move the extra text to next line, I have updated maxLines from 1 to 2 to allow the extra text to move to the next line. This is with assumption that the translated text can never be more than 2 lines. I have reverted changes on end and start margins as this will not work and the device's screen gets thinner line wrapping would work better.

BenHenning commented:
> @BenHenning it actually makes sense to move the extra text to next line, I have updated maxLines from 1 to 2 to allow the extra text to move to the next line. This is with assumption that the translated text can never be more than 2 lines. I have reverted changes on end and start margins as this will not work and the device's screen gets thinner line wrapping would work better.

Thanks @KevinGitonga. Can you also update the screenshots in the PR? It's hard to verify the code just by looking at it, so the screenshots are really important to also convey the difference from a user's perspective.

I'm also assigning @rt4914 so that he can leave his thoughts.

KevinGitonga committed:
Update maxLines for other portrait layouts.

rt4914 commented on app/src/main/res/layout/onboarding_slide_final.xml:
```diff
@@ -45,7 +45,7 @@
         android:fontFamily="sans-serif-medium"
         android:gravity="center_horizontal"
-        android:maxLines="1"
+        android:maxLines="2"
```
I think we should remove this `maxLines` attributes completely for this screen. So that even if in some screen it goes to three-lines the UI will be visible fully.

KevinGitonga replied:
Changes Applied.

rt4914 reviewed (changes requested):
@KevinGitonga Please update your PR title and description as you are not making any change to margins.

Thanks.

KevinGitonga commented:
> @KevinGitonga Please update your PR title and description as you are not making any change to margins.
> 
> Thanks.

Hey @rt4914 have made the changes as requested PTAL.

KevinGitonga committed:
Make changes as requested on review.

rt4914 commented on app/src/main/res/layout-sw600dp-port/onboarding_slide.xml:
```diff
@@ -41,7 +41,7 @@
         android:layout_marginEnd="160dp"
         android:gravity="center_horizontal"
-        android:maxLines="1"
+        android:maxLines="2"
```
Same previous comment applies to this code too.

Remove maxLines attribute.

Also, please re-generate screenshots and update PR description just to be sure

KevinGitonga replied:
Noted.

rt4914 commented on app/src/main/res/layout/onboarding_slide.xml:
```diff
@@ -43,7 +43,7 @@
         android:fontFamily="sans-serif-medium"
         android:gravity="center_horizontal"
-        android:maxLines="1"
+        android:maxLines="2"
```
Same previous comment applies to this code too.

Remove maxLines attribute.

Also, please re-generate screenshots and update PR description just to be sure

KevinGitonga replied:
Noted.

rt4914 commented on app/src/main/res/layout-sw600dp-port/onboarding_slide_final.xml:
```diff
@@ -43,7 +43,7 @@
         android:fontFamily="sans-serif-medium"
         android:gravity="center_horizontal"
-        android:maxLines="1"
+        android:maxLines="2"
```
Same previous comment applies to this code too.

Remove maxLines attribute.

Also, please re-generate screenshots and update PR description just to be sure

KevinGitonga replied:
Noted.

rt4914 reviewed (changes requested):
@KevinGitonga PTAL

KevinGitonga committed:
Make more changes as advised by reviewer.

KevinGitonga commented:
@rt4914 have made changes. For phone screenshots they are similar to those shared and for the tablet also they remain as i had updated. This is after checking on my device and tablet emulator on both Portrait and Landscape. PTAL.

rt4914 reviewed (approved):
LGTM, Thanks.

rt4914 commented:
@KevinGitonga CI checks are failing. Have a look.

KevinGitonga committed:
Merge branch 'oppia:develop' into fix_portuguese_text_cut_off

BenHenning reviewed (approved):
Thanks @KevinGitonga! This also LGTM.

BenHenning commented:
Except one thing @KevinGitonga: please make sure to update the title to include the bug being fixed (as I edited just now) for future PRs.

KevinGitonga commented:
> Except one thing @KevinGitonga: please make sure to update the title to include the bug being fixed (as I edited just now) for future PRs.

Thank's @BenHenning well noted.
